data/filedefs/filetypes.HCL.conf

```diff
@@ -0,0 +1,40 @@
+# For complete documentation of this file, please see Geany's main documentation
+[styling=C]
+# style "preprocessor" as comment, as for HCL lines starting with "#" are comments
+preprocessor=comment
+
+[keywords]
+# all items must be in one line
+primary=data local locals module provider resource terraform var
+# for autocompletion tags, see https://wiki.geany.org/tags/start#hcl_terraformopentofu_tags
+
+[lexer_properties=C]
+styling.within.preprocessor=0
```

Comment:
An explanation of the relation to the comment hack would be welcome here

Reply:
Luckily, we don't need this hack any longer :).